pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -199,19 +325,105 @@ func sendCertsProtobuf(attestReq *attest.ZAttestReq, iteration int) {
 	_, _, rtf, err := zedcloud.SendOnAllIntf(&zedcloudCtx, attestURL,
 		size, buf, iteration, bailOnHTTPErr)
 	if err != nil {
+		// Hopefully next timeout will be more successful
 		if rtf == types.SenderStatusRemTempFail {
-			log.Errorf("sendCertsProtobuf remoteTemporaryFailure: %s",
+			log.Errorf("sendEveNodeCertsProtobuf remoteTemporaryFailure: %s",
 				err)
 		} else {
-			log.Errorf("sendCertsProtobuf failed: %s", err)
+			log.Errorf("sendEveNodeCertsProtobuf failed: %s", err)
 		}
-		// Try sending later
-		// The buf might have been consumed
-		buf := bytes.NewBuffer(data)
-		if buf == nil {
-			log.Fatal("malloc error")
-		}
-		zedcloud.SetDeferred("attest:"+deviceUUID, buf, size, attestURL,
+		zedcloud.SetDeferred("attest:"+reqID, buf, size, attestURL,
 			zedcloudCtx, true)
 	}
 }
+
+// initialize cipher pubsub trigger handlers and channels
+func cipherModuleInitialize(ctx *zedagentContext, ps *pubsub.PubSub) {
+
+	// create the trigger channels
+	ctx.cipherCtx.triggerEveNodeCerts = make(chan struct{}, 1)
+	ctx.cipherCtx.triggerControllerCerts = make(chan struct{}, 1)
+}
+
+// start the task threads
+func cipherModuleStart(ctx *zedagentContext) {
+	if !zedcloud.UseV2API() {
+		log.Infof("V2 APIs are still not enabled")
+		// we will run the tasks for watchdog
+	}
+	// start the eve node certificate push task
+	go eveNodeCertsTask(ctx, ctx.cipherCtx.triggerEveNodeCerts)
+
+	// start the controller certificate fetch task
+	go controllerCertsTask(ctx, ctx.cipherCtx.triggerControllerCerts)
+}
+
+// Controller certificate, check whether there is a Sha mismatch
+// to trigger the post request
+func handleControllerCertsSha(ctx *zedagentContext,
+	config *zconfig.EdgeDevConfig) {
+	sumHash := hex.EncodeToString(ctx.cipherCtx.cfgControllerCertHash)
+	certHash := config.GetControllercertConfighash()
```

Comment:
If this is not set by the controller (I think that means you get an empty string), then we shouldn't trigger because that just wastes bandwidth.
Instead you can log that the ControllercertConfigHash is not set.

Reply:
sure, will do.